File: backend/src/models/knowledge.py
"""Knowledge API DTOs aligned with docs/api-contracts.md."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

KnowledgeQaStatus = Literal["enabled", "disabled"]
KnowledgeDocumentStatus = Literal["processing", "completed", "failed"]
KnowledgeRecentType = Literal["qa", "document"]
KnowledgeEmbeddingStatus = Literal["completed", "fallback"]


class KnowledgeOverviewResponseData(BaseModel):
    qa_count: int
    enabled_qa_count: int
    document_count: int
    completed_document_count: int
    latest_updated_at: str


class KnowledgeRecentItem(BaseModel):
    type: KnowledgeRecentType
    id: str
    title: str
    status: KnowledgeQaStatus | KnowledgeDocumentStatus
    updated_at: str


class KnowledgeRecentResponseData(BaseModel):
    items: list[KnowledgeRecentItem]


class KnowledgeQaRead(BaseModel):
    id: str
    question: str
    answer: str
    status: KnowledgeQaStatus
    created_at: str
    updated_at: str


class KnowledgeQaListResponseData(BaseModel):
    items: list[KnowledgeQaRead]
    total: int
    page: int
    page_size: int


class KnowledgeQaCreate(BaseModel):
    question: str
    answer: str
    status: KnowledgeQaStatus


class KnowledgeQaUpdate(BaseModel):
    question: str
    answer: str
    status: KnowledgeQaStatus


class KnowledgeQaMutationResponseData(BaseModel):
    qa: KnowledgeQaRead
    embedding_status: KnowledgeEmbeddingStatus


class KnowledgeQaDeleteResponseData(BaseModel):
    deleted: bool
    qa_id: str

File: frontend/src/types/knowledge.ts
export type KnowledgeQaStatus = 'enabled' | 'disabled'
export type KnowledgeDocumentStatus = 'processing' | 'completed' | 'failed'
export type KnowledgeRecentType = 'qa' | 'document'

export interface KnowledgeOverviewResponseData {
  qa_count: number
  enabled_qa_count: number
  document_count: number
  completed_document_count: number
  latest_updated_at: string
}

export interface KnowledgeRecentItemDto {
  type: KnowledgeRecentType
  id: string
  title: string
  status: KnowledgeQaStatus | KnowledgeDocumentStatus
  updated_at: string
}

export interface KnowledgeRecentRequest {
  limit?: number
}

export interface KnowledgeRecentResponseData {
  items: KnowledgeRecentItemDto[]
}

export interface KnowledgeQaDto {
  id: string
  question: string
  answer: string
  status: KnowledgeQaStatus
  created_at: string
  updated_at: string
}

export interface KnowledgeQaListRequest {
  keyword?: string
  status?: KnowledgeQaStatus
  page?: number
  page_size?: number
}

export interface KnowledgeQaListResponseData {
  items: KnowledgeQaDto[]
  total: number
  page: number
  page_size: number
}

export interface CreateKnowledgeQaRequest {
  question: string
  answer: string
  status: KnowledgeQaStatus
}

export interface UpdateKnowledgeQaRequest {
  question: string
  answer: string
  status: KnowledgeQaStatus
}

export interface CreateKnowledgeQaResponseData {
  qa: KnowledgeQaDto
  embedding_status: 'completed' | 'fallback'
}

export interface UpdateKnowledgeQaResponseData {
  qa: KnowledgeQaDto
  embedding_status: 'completed' | 'fallback'
}

export interface DeleteKnowledgeQaResponseData {
  deleted: boolean
  qa_id: string
}
